medusa/clients/torrent/generic.py

```diff
@@ -44,67 +44,30 @@ def __init__(self, name, host=None, username=None, password=None):
         self.username = app.TORRENT_USERNAME if username is None else username
         self.password = app.TORRENT_PASSWORD if password is None else password
         self.host = app.TORRENT_HOST if host is None else host
+        self.torrent_path = app.TORRENT_PATH if torrent_path is None else torrent_path
         self.rpcurl = app.TORRENT_RPCURL
         self.url = None
         self.response = None
         self.auth = None
-        self.last_time = time.time()
+        self.message = None
         self.session = ClientSession()
         self.session.auth = (self.username, self.password)
-
-    def _request(self, method='get', params=None, data=None, files=None, cookies=None):
-
-        if time.time() > self.last_time + 1800 or not self.auth:
```

Comment:
what was your reasoning to remove the authentication caching?

Reply:
Because of the fact that the class in not used as an object but as a bunch of functions, every time a function of that GenericClient class is called, the __init__ function is called first, so all the self variables are initialized initialized fresh and that makes this attempt of caching useless.
Instead I added a call to self._get_auth in the __init__ to be sure the authorisation is done before every call.
If you want caching then you have to convert it to a real object so the __ini__ is call only once and the self variables are kept alive.